src/index.js

```diff
@@ -122,18 +123,30 @@ class Service {
     const where = Object.assign({}, filter(params.query || {}).query);
     const mapIds = page => page.data.map(current => current[this.id]);
 
-    // By default we will just query for the one id. For multi patch
-    // we create a list of the ids of all items that will be changed
-    // to re-query them after the update
-    const ids = id === null ? this._find(params)
-        .then(mapIds) : Promise.resolve([ id ]);
-
     if (id !== null) {
       where[this.id] = id;
     }
 
     const options = Object.assign({}, params.sequelize, { where });
 
+    // This is the best way to implement patch in sql, the other dialects 'should' use a transaction.
+    if (this.dialect === 'postgres') {
```

Comment:
I'd check `options.Model.sequelize.options.dialect` here and remove setting it in the constructor